src/_pytest/fixtures.py

```diff
@@ -811,6 +813,8 @@ def __init__(
         self.has_location = baseid is not None
         self.func = func
         self.argname = argname
+        if callable(scope):
+            scope = scope(self._fixturemanager.config)
```

Comment:
if it happens at that level, its not of much value

a key part is how it would be able to determine it per test item

Reply:
Hmm, @nicoddemus said that the goal is to evaluate the scope function only once. Did I misunderstand something?

Reply:
Here's the original intent from https://github.com/pytest-dev/pytest/issues/1682#issue-162492156:

> The idea is that developers might be interested to have different scopes for a fixture depending on a command line option, so that one might choose to change a fixture scope to session to avoid costly initialization during development, but opt to change it to function in a C.I. server.

Reply:
indeed, that one was my mistake, i had something else in mind

i apologize for the confusion i caused